website/notifications/events/model.py

```diff
@@ -0,0 +1,352 @@
+# -*- coding: utf-8 -*-
+"""Basic Event handling for events that need subscriptions"""
+
+from furl import furl
+from datetime import datetime
+from six import add_metaclass
+
+from website.notifications import emails
+from website.notifications.constants import NOTIFICATION_TYPES
+from website.models import Node
+from website.notifications.events import utils as event_utils
+from website.notifications import utils
+
+
+class _EventMeta(type):
```

Comment:
Discussed in person: we can get rid of this metaclass and properly represent the relationship between log actions (e.g. `NodeLog.FILE_ADDED` and `Event` types by using a simple dictionary.

```python
event_map = {
    NodeLog.FILE_ADDED: FileAdded
}
```

If you wanted, you could implement a decorator for convenience:

```python
@register_event(NodeLog.FILE_ADDED)
class FileAdded(FileEvent):
    # ...
```